src/components/utils.ts

```diff
@@ -74,6 +74,9 @@ export const getAlertsAndRules = (
   // Add `rule` object to each alert
   const alerts = _.flatMap(rules, (rule) => rule.alerts.map((a) => ({ rule, ...a })));
 
+  // `alert.rule.labels` contains external labels which we want to include with `alert.labels`.
+  alerts.map((alert) => (alert.labels = { ...alert.labels, ...alert.rule.labels }));
```

Comment:
```suggestion
  alerts.map((alert) => (alert.labels = { ...alert.rule.labels, ...alert.labels }));
```

If the rule and the alert have a label with the same key, I think we want the alert's label to be preserved.

Reply:
I see what you're saying now. Yes, that makes sense, I'll update this in the next commit.